File: samples/go-project/pkg/edge_cases.go
package pkg

// Edge case: comment should not appear as a caller
// FormatOutput("this is a comment, not a call")

func commentedOutCall() {
}

// Edge case: same-name function should not appear as caller of utils.FormatOutput
func FormatOutput(localArg string) string {
	return "local: " + localArg
}

func callsLocalFormatOutput() {
	FormatOutput("local only")
}

File: samples/java-project/src/main/java/com/example/EdgeCases.java
package com.example;

// Edge case: comment should not appear as a caller
// StringUtils.capitalize("this is a comment, not a call")

class EdgeCases {
    void commentedOutCall() {
    }

    // Edge case: same-name method should not appear as caller of StringUtils.capitalize
    String capitalize(String localArg) {
        return "local: " + localArg;
    }

    void callsLocalCapitalize() {
        capitalize("local only");
    }
}
